packages/sso/src/oidc/discovery.test.ts

```diff
@@ -326,18 +350,232 @@ describe("OIDC Discovery", () => {
+					details: {
+						endpoint: "jwks_uri",
+						url: "https://idp.example.com/.well-known/jwks.json",
+					},
+				}),
+			);
+
+			expect(() =>
+				normalizeDiscoveryUrls(
+					doc,
+					"https://idp.example.com",
+					(url) => !url.endsWith("/userinfo"),
+				),
+			).toThrowError(
+				expect.objectContaining({
+					code: "discovery_untrusted_origin",
+					message:
+						'The userinfo_endpoint "https://idp.example.com/userinfo" is not trusted by your trusted origins configuration.',
+					details: {
+						endpoint: "userinfo_endpoint",
+						url: "https://idp.example.com/userinfo",
+					},
+				}),
+			);
+
+			expect(() =>
+				normalizeDiscoveryUrls(
+					doc,
+					"https://idp.example.com",
+					(url) => !url.endsWith("/revoke"),
+				),
+			).toThrowError(
+				expect.objectContaining({
+					code: "discovery_untrusted_origin",
+					message:
+						'The revocation_endpoint "https://idp.example.com/revoke" is not trusted by your trusted origins configuration.',
+					details: {
+						endpoint: "revocation_endpoint",
+						url: "https://idp.example.com/revoke",
+					},
+				}),
+			);
+		});
 	});
 
-	describe("normalizeUrl (stub)", () => {
-		it("should return endpoint unchanged in Phase 1", () => {
+	describe("normalizeUrl", () => {
+		it("should return endpoint unchanged if already absolute", () => {
 			const endpoint = "https://idp.example.com/oauth2/token";
-			expect(normalizeUrl(endpoint, "https://idp.example.com")).toBe(endpoint);
+			expect(normalizeUrl("url", endpoint, "https://idp.example.com")).toBe(
+				endpoint,
+			);
+		});
+
+		it("should return endpoint as an absolute url", () => {
+			const endpoint = "/oauth2/token";
+			expect(normalizeUrl("url", endpoint, "https://idp.example.com")).toBe(
+				"https://idp.example.com/oauth2/token",
+			);
+		});
+
+		it("should return endpoint as an absolute url preserving the issuer base path", () => {
+			const endpoint = "/oauth2/token";
+			expect(
+				normalizeUrl("url", endpoint, "https://idp.example.com/base"),
+			).toBe("https://idp.example.com/base/oauth2/token");
+		});
+
+		it("should return endpoint as an absolute url preserving the issuer base path", () => {
+			const endpoint = "oauth2/token";
+			expect(
+				normalizeUrl("url", endpoint, "https://idp.example.com/base"),
+			).toBe("https://idp.example.com/base/oauth2/token");
+		});
+
+		it("should return endpoint as an absolute url preserving the issuer base path (trailing slash)", () => {
+			const endpoint = "/oauth2/token";
+			expect(
+				normalizeUrl("url", endpoint, "https://idp.example.com/base/"),
+			).toBe("https://idp.example.com/base/oauth2/token");
+		});
+
+		it("should return endpoint as an absolute url preserving the issuer base path (multiple slashes)", () => {
+			const endpoint = "//oauth2/token";
+			expect(
+				normalizeUrl("url", endpoint, "https://idp.example.com/base//"),
+			).toBe("https://idp.example.com/base/oauth2/token");
+		});
```

Comment:
This are great test scenarios , what do you think reducing it (as almost same test) with this: 
```suggestion
		it.each([
			["/oauth2/token", "https://idp.example.com/base", "endpoint with leading slash"],
			["oauth2/token", "https://idp.example.com/base", "endpoint without leading slash"],
			["/oauth2/token", "https://idp.example.com/base/", "issuer with trailing slash"],
			["//oauth2/token", "https://idp.example.com/base//", "multiple slashes"],
		])(
			"should resolve relative endpoint preserving issuer base path (%s, %s) - %s",
			(endpoint, issuer) => {
				expect(normalizeUrl("url", endpoint, issuer)).toBe(
					"https://idp.example.com/base/oauth2/token",
				);
			},
		);
		
```

Reply:
To be honest I am not a big fan of introducing any kind of logic or iteration to tests 😅 , it just makes them harder to reason about and debug.